test/NetEscapades.AspNetCore.SecurityHeaders.Test/ReportingEndpointsHeaderBuilderTests.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using FluentAssertions;
+using NetEscapades.AspNetCore.SecurityHeaders.Headers;
+using Xunit;
+
+namespace NetEscapades.AspNetCore.SecurityHeaders.Test;
+
+public class ReportingEndpointsHeaderBuilderTests
+{
+    [Fact]
+    public void Build_WhenNoValues_ReturnsNull()
+    {
+        var builder = new ReportingEndpointsHeaderBuilder();
+
+        var result = builder.Build();
+        result.Should().BeNullOrEmpty();
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("")]
+    public void Build_WhenNullName_Throws(string name)
+    {
+        var builder = new ReportingEndpointsHeaderBuilder();
+
+        Assert.Throws<ArgumentNullException>(() => builder.AddEndpoint(name, "http://localhost:80/"));
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("")]
+    public void Build_WhenNullUrl_Throws(string url)
+    {
+        var builder = new ReportingEndpointsHeaderBuilder();
+
+        Assert.Throws<ArgumentNullException>(() => builder.AddDefaultEndpoint(url));
+    }
+
+    [Theory]
+    [InlineData("not/\\aurl")]
+    [InlineData("/relative")]
+    public void Build_WhenInvalidUrlUrl_Throws(string url)
```

Comment:
@andrewlock I think relative urls should be allowed.

I can tell from experience that they work.

And the standard says that the value is resolved against response's url.
https://www.w3.org/TR/reporting-1/#process-header in 5.2.:
> Let endpoint url be the result of executing the URL parser on endpoint url string, with base URL set to response’s url. If endpoint url is failure, then continue.

Reply:
Hmmm, I could have _sworn_ I read that it had to be an absolute URL 🤦‍♂️ I think I must have read [path-absolute-URL string](https://url.spec.whatwg.org/#path-absolute-url-string) for absolute URL.

I agree, I'll remove that limitation. Makes it easier on this side anyway 😉 

Thanks!